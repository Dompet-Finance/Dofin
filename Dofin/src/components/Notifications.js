'use strict'

import React from 'react'

import {
  View,
  Text,
  Button,
} from 'react-native'

class AppBar extends React.Component {
  render() {
    return (
      <View>
        <Text>
          AppBar
        </Text>
      </View>
    )
  }
}

class NotificationList extends React.Component {
  render() {
    return (
      <View>
        <Text>
          Notification List
        </Text>
      </View>
    )
  }
}

export class Transactions extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      income: 0,
      expense: 0,
    }
  }

  render() {
    return (
      <View>
        <AppBar />
        <NotificationList />
        { this.state.expense > this.state.income
          &&
          (<Text>
            Overlimit
          </Text>)
        }
      </View>
    )
  }
}

const styles = {
}

export default Transactions