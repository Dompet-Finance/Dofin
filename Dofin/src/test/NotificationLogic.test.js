'use strict'

import 'react-native'
import 'react-dom'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Notifications from '../components/Notifications.js'

describe('Test Notification Logic component', () => {

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Notifications />)
  })

  test('renders correctly', () => {
    const tree = renderer.create(
      <Notifications />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('should show up overlimit warning in notification', () => {
    const fakeState = {
      income: 1000,
      expense: 2000,
    }
    wrapper.setState(fakeState)
    
    expect(wrapper).toMatchSnapshot();
  })

})

