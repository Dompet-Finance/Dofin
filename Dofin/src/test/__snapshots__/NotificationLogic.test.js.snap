// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Notification Logic component renders correctly 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      AppBar
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Notification List
    </Text>
  </View>
</View>
`;

exports[`Test Notification Logic component should show up overlimit warning in notification 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <AppBar />
    <NotificationList />
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Overlimit
    </Text>
</View>,
  "nodes": Array [
    <View>
      <AppBar />
      <NotificationList />
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
      >
            Overlimit
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Transactions />,
      "_debugID": 18,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Transactions {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "expense": 2000,
          "income": 1000,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <AppBar />
          <NotificationList />
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
          >
                    Overlimit
          </Text>
</View>,
        "_debugID": 19,
        "_renderedOutput": <View>
          <AppBar />
          <NotificationList />
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
          >
                    Overlimit
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Transactions />,
}
`;
